Lib/test/test_unpack.py

```diff
@@ -167,5 +220,49 @@ def test_extended_oparg_not_ignored(self):
             y = unpack_400(range(400))
             self.assertEqual(y, 399)
 
+    def test_exception_context_when_len_fails(self):
+        """When `__len__()` raises an Exception, preserve exception context"""
+        code = textwrap.dedent(
+            """
+            class CustomSeq:
+                def __len__(self):
+                    raise RuntimeError
+
+                def __getitem__(self, i):
+                    return i*2
+
+            x, y, z = CustomSeq()
+            """
+        )
+        with self.assertRaises(UnpackError) as cm:
+            exec(code)
+
+        traceback_text = ''.join(traceback.format_exception(cm.exception))
+        self.assertIn(
+            "RuntimeError\n\n"
+            "During handling of the above exception, another exception occurred:\n\n"
+            "UnpackError: too many values to unpack (expected 3)\n",
+            traceback_text,
+        )
+
+    def test_baseexception_propagation_when_len_fails(self):
+        """if `__len__()` raises a `BaseException`, propagate that as-is"""
```

Comment:
```suggestion
        """Propoagate any BaseException raised in __len__()"""
```

I would personally propagate *any* exception in `__len__`.

Reply:
I think we went through that already.
IMO, since getting the length is not an operation the user asked for. It's fine to ignore `Exception`s in it. But not `BaseException` (like Crtl+C).

Reply:
(Yes, it's just that I commented the files as I saw them and only noticed at the end; should have removed this comment in particular).